libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -3763,7 +3807,10 @@ instance BaseUniverse fn => Functions (SetFn fn) fn where
         | NilCtx HOLE <- ctx
         , Evidence <- prerequisites @fn @[a] ->
             case spec of
-              MemberSpec [xs] -> typeSpec $ ListSpec Nothing (Set.toList xs) TrueSpec (MemberSpec $ Set.toList xs) NoFold
+              MemberSpec es
+                | NE.length es == 1
+                , Just xs <- NE.nonEmpty (Set.toList (NE.head es)) ->
+                    typeSpec $ ListSpec Nothing (NE.toList xs) TrueSpec (MemberSpec xs) NoFold
```

Comment:
not semantics preserving

Reply:
So how about this
case spec of
              MemberSpec (xs :| []) -> 
                typeSpec $ ListSpec Nothing (Set.toList xs) TrueSpec (MemberSpec $ Set.toList xs) NoFold